lib/services/media_repository.dart

```diff
@@ -14,4 +14,6 @@ mixin MediaRepository {
   Future<String?> saveSamplesToWaveFile(String basename, Float64List samples);
 
   Future<void> delete(String relativeFilePath);
+
+  Future<void> deleteTemporaryFiles(String absoluteSourceFilePath);
```

Comment:
Move this function to file system and rename function to:

```suggestion
  Future<void> deleteIfTmpFile(String absoluteSourceFilePath);
```